### ISSUE TITLE
[FEA] Support parse_url function

### ISSUE THREAD

**Issue by nvliyuan** [NVIDIA/spark-rapids#8512]
It would be better to support [parse_url](https://spark.apache.org/docs/latest/api/sql/index.html#parse_url) function:
Codes:
```
scala> spark.sql("SELECT parse_url('http://spark.apache.org/path?query=1', 'HOST')").show
```
driverlogs:
```
@Expression <Alias> parse_url(http://spark.apache.org/path?query=1, HOST, false) AS parse_url(http://spark.apache.org/path?query=1, HOST)#3 could run on GPU
    ! <ParseUrl> parse_url(http://spark.apache.org/path?query=1, HOST, false) cannot run on GPU because GPU does not currently support the operator class org.apache.spark.sql.catalyst.expressions.ParseUrl
```

**Comment by NVnavkumar** [comment]
This is a duplicate of https://github.com/NVIDIA/spark-rapids/issues/6969

**Comment by jlowe** [comment]
@sameerz why was this closed?  #8761 was not merged, and #9481 is still pending.

**Comment by NVnavkumar** [comment]
> @sameerz why was this closed? #8761 was not merged, and #9481 is still pending.

It's a dupe of https://github.com/NVIDIA/spark-rapids/issues/6969
